src/Surfnet/StepupRa/RaBundle/EventListener/LocaleListener.php

```diff
@@ -18,48 +18,36 @@
 
 namespace Surfnet\StepupRa\RaBundle\EventListener;
 
-use Surfnet\StepupMiddlewareClientBundle\Identity\Dto\Identity;
+use Surfnet\StepupBundle\Service\LocaleProviderService;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\HttpKernel\Event\GetResponseEvent;
 use Symfony\Component\HttpKernel\KernelEvents;
-use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
 use Symfony\Component\Translation\TranslatorInterface;
 
 final class LocaleListener implements EventSubscriberInterface
 {
     /**
-     * @var \Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface
+     * @var LocaleProviderService
      */
-    private $tokenStorage;
+    private $localeProviderService;
 
     /**
      * @var TranslatorInterface
      */
     private $translator;
 
-    public function __construct(TokenStorageInterface $tokenStorage, TranslatorInterface $translator)
+    public function __construct(LocaleProviderService $localeProviderService, TranslatorInterface $translator)
     {
-        $this->tokenStorage = $tokenStorage;
+        $this->localeProviderService = $localeProviderService;
         $this->translator = $translator;
     }
 
     public function setRequestLocale(GetResponseEvent $event)
     {
-        $token = $this->tokenStorage->getToken();
-
-        if (!$token) {
-            return;
-        }
-
-        /** @var Identity $identity */
-        $identity = $token->getUser();
-
-        if (!$identity instanceof Identity) {
-            return;
-        }
+        $preferredLocale = $this->localeProviderService->determinePreferredLocale();
```

Comment:
I don't think this was needed; this interface was designed for use by `LocaleCookieListener`, to aid in integrating the bundle into this application. Its reasons to change lie with the bundle, and when it changes it suddenly impacts more in this application than expected. While it is unlikely that will really become an issue, because this is all pretty simple (currently), it's something to keep in mind. Because you changed this neatly in its own commit, it would be quite simple to pull out again, but I'll leave that to you to decide.